### ISSUE TITLE
Frequent hangs in ves_icall_System_Net_Dns_GetHostByName_internal (getaddrinfo)

### ISSUE THREAD

**Issue by alexischr** [mono/mono#7935]
### Example
```
***** MonoTests.System.Net.Http.HttpClientTest.ModifyHandlerAfterFirstRequest
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Net.Dns.GetHostByName_internal (string,string&,string[]&,string[]&,int) <0x00015>
  at System.Net.Dns.GetHostByName (string) [0x0000e] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System/System.Net/Dns.cs:439
  at System.Net.EndPointManager.GetEPListener (string,int,System.Net.HttpListener,bool) [0x0001f] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System/System.Net/EndPointManager.cs:88
  at System.Net.EndPointManager.RemovePrefixInternal (string,System.Net.HttpListener) [0x0002d] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System/System.Net/EndPointManager.cs:156
  at System.Net.EndPointManager.RemoveListener (System.Net.HttpListener) [0x00024] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System/System.Net/EndPointManager.cs:135
  at System.Net.HttpListener.Close (bool) [0x00006] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System/System.Net/HttpListener.cs:217
  at System.Net.HttpListener.Abort () [0x00012] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System/System.Net/HttpListener.cs:197
  at MonoTests.System.Net.Http.HttpClientTest.ModifyHandlerAfterFirstRequest () [0x00079] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System.Net.Http/Test/System.Net.Http/HttpClientTest.cs:1411
  at (wrapper runtime-invoke) object.runtime_invoke_void__this__ (object,intptr,intptr,intptr) [0x0004f] in <7815783a5a844a60be9cf9a4fd1fe1f9>:0
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&) <0x00012>
  at System.Reflection.MonoMethod.Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo) [0x0003b] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/corlib/System.Reflection/MonoMethod.cs:305
  at System.Reflection.MethodBase.Invoke (object,object[]) [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/reflection/methodbase.cs:229
  at NUnit.Framework.Internal.Reflect.InvokeMethod (System.Reflection.MethodInfo,object,object[]) [0x00009] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/Reflect.cs:215
  at NUnit.Framework.Internal.Commands.TestMethodCommand.RunNonAsyncTestMethod (NUnit.Framework.Internal.TestExecutionContext) [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/Commands/TestMethodCommand.cs:114
  at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod (NUnit.Framework.Internal.TestExecutionContext) [0x0001a] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/Commands/TestMethodCommand.cs:90
  at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute (NUnit.Framework.Internal.TestExecutionContext) [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/Commands/TestMethodCommand.cs:66
  at NUnit.Framework.Internal.Commands.SetUpTearDownCommand.Execute (NUnit.Framework.Internal.TestExecutionContext) [0x00007] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/Commands/SetUpTearDownCommand.cs:84
  at NUnit.Framework.Internal.Commands.FlakyTestRetriesCommand.Execute (NUnit.Framework.Internal.TestExecutionContext) [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/Commands/FlakyTestRetriesCommand.cs:73
  at NUnit.Framework.Internal.WorkItems.SimpleWorkItem.PerformWork () [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/SimpleWorkItem.cs:64
  at NUnit.Framework.Internal.WorkItems.WorkItem.DispatchWork (object) [0x00077] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:205
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00071] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:961
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:908
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object) [0x0002b] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:897
  at NUnit.Framework.Internal.WorkItems.WorkItem.RunTest () [0x00005] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:188
  at NUnit.Framework.Internal.WorkItems.WorkItem.Execute (NUnit.Framework.Internal.TestExecutionContext) [0x00063] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:136
  at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.RunChildren () [0x00035] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/CompositeWorkItem.cs:154
  at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.PerformWork () [0x000de] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/CompositeWorkItem.cs:93
  at NUnit.Framework.Internal.WorkItems.WorkItem.DispatchWork (object) [0x00077] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:205
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00071] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:961
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:908
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object) [0x0002b] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:897
  at NUnit.Framework.Internal.WorkItems.WorkItem.RunTest () [0x00005] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:188
  at NUnit.Framework.Internal.WorkItems.WorkItem.Execute (NUnit.Framework.Internal.TestExecutionContext) [0x00063] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:136
  at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.RunChildren () [0x00035] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/CompositeWorkItem.cs:154
  at NUnit.Framework.Internal.WorkItems.CompositeWorkItem.PerformWork () [0x000de] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/CompositeWorkItem.cs:93
  at NUnit.Framework.Internal.WorkItems.WorkItem.DispatchWork (object) [0x00077] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:205
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00071] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:961
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00000] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:908
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object) [0x0002b] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:897
  at NUnit.Framework.Internal.WorkItems.WorkItem.RunTest () [0x00005] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:188
  at NUnit.Framework.Internal.WorkItems.WorkItem.Execute (NUnit.Framework.Internal.TestExecutionContext) [0x00063] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/WorkItems/WorkItem.cs:136
  at NUnit.Framework.Internal.NUnitLiteTestAssemblyRunner.Run (NUnit.Framework.Api.ITestListener,NUnit.Framework.Api.ITestFilter) [0x0005a] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Internal/NUnitLiteTestAssemblyRunner.cs:136
  at NUnitLite.Runner.TextUI.RunTests (NUnit.Framework.Api.ITestFilter) [0x0001e] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Runner/TextUI.cs:308
  at NUnitLite.Runner.TextUI.Execute (string[]) [0x002c4] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/external/nunit-lite/NUnitLite-1.0.0/src/framework/Runner/TextUI.cs:210
  at NUnitLite.Console.Main (string[]) [0x00005] in /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/tools/nunit-lite/nunit-lite-console/Console.cs:79
  at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) [0x00057] in <c149c17ae28043f98b925378cca9be13>:0

Native stacktrace:

	0   mono                                0x0016e209 mono_handle_native_crash + 233
	1   mono                                0x001d4679 sigabrt_signal_handler + 153
	2   libsystem_platform.dylib            0xa1738e5b _sigtramp + 43
	3   ???                                 0xffffffff 0x0 + 4294967295
	4   libsystem_info.dylib                0xa162416d mdns_addrinfo + 474
	5   libsystem_info.dylib                0xa1623f6f search_addrinfo + 123
	6   libsystem_info.dylib                0xa16205bd si_addrinfo + 1838
	7   libsystem_info.dylib                0xa161fde0 getaddrinfo + 145
	8   mono                                0x00380873 mono_get_address_info + 179
	9   mono                                0x002ab0d5 ves_icall_System_Net_Dns_GetHostByName_internal + 197
	10  ???                                 0x0359af8c 0x0 + 56209292
	11  ???                                 0x0359acfc 0x0 + 56208636
	12  ???                                 0x0359a410 0x0 + 56206352
	13  ???                                 0x036009cc 0x0 + 56625612
	14  ???                                 0x03600864 0x0 + 56625252
	15  ???                                 0x036006d0 0x0 + 56624848
	16  ???                                 0x036006a0 0x0 + 56624800
	17  ???                                 0x036075f0 0x0 + 56653296
	18  ???                                 0x02ca53a3 0x0 + 46814115
	19  mono                                0x000c3e4b mono_jit_runtime_invoke + 843
	20  mono                                0x0029e31a do_runtime_invoke + 74
	21  mono                                0x002a2066 mono_runtime_try_invoke_array + 1350
	22  mono                                0x002a2545 mono_runtime_invoke_array_checked + 37
	23  mono                                0x0024fa20 ves_icall_InternalInvoke + 704
	24  ???                                 0x02ca626c 0x0 + 46817900
	25  mscorlib.dll.dylib                  0x01b17734 System_Reflection_MonoMethod_Invoke_object_System_Reflection_BindingFlags_System_Reflection_Binder_object___System_Globalization_CultureInfo + 164
	26  mscorlib.dll.dylib                  0x01b072b7 System_Reflection_MethodBase_Invoke_object_object__ + 71
	27  ???                                 0x02ca5fcc 0x0 + 46817228
	28  ???                                 0x02ca5f78 0x0 + 46817144
	29  ???                                 0x02ca5ed8 0x0 + 46816984
	30  ???                                 0x02ca5dfc 0x0 + 46816764
	31  ???                                 0x02ca5c0b 0x0 + 46816267
	32  ???                                 0x02ca57d6 0x0 + 46815190
	33  ???                                 0x02ca5749 0x0 + 46815049
	34  ???                                 0x02ca1970 0x0 + 46799216
	35  mscorlib.dll.dylib                  0x01a4c37f System_Threading_ExecutionContext_RunInternal_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 383
	36  mscorlib.dll.dylib                  0x01a4c1f4 System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 52
	37  mscorlib.dll.dylib                  0x01a4c165 System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object + 85
	38  ???                                 0x02ca17dc 0x0 + 46798812
	39  ???                                 0x02ca1574 0x0 + 46798196
	40  ???                                 0x02ca413e 0x0 + 46809406
	41  ???                                 0x02ca320c 0x0 + 46805516
	42  ???                                 0x02ca1970 0x0 + 46799216
	43  mscorlib.dll.dylib                  0x01a4c37f System_Threading_ExecutionContext_RunInternal_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 383
	44  mscorlib.dll.dylib                  0x01a4c1f4 System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 52
	45  mscorlib.dll.dylib                  0x01a4c165 System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object + 85
	46  ???                                 0x02ca17dc 0x0 + 46798812
	47  ???                                 0x02ca1574 0x0 + 46798196
	48  ???                                 0x02ca413e 0x0 + 46809406
	49  ???                                 0x02ca320c 0x0 + 46805516
	50  ???                                 0x02ca1970 0x0 + 46799216
	51  mscorlib.dll.dylib                  0x01a4c37f System_Threading_ExecutionContext_RunInternal_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 383
	52  mscorlib.dll.dylib                  0x01a4c1f4 System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object_bool + 52
	53  mscorlib.dll.dylib                  0x01a4c165 System_Threading_ExecutionContext_Run_System_Threading_ExecutionContext_System_Threading_ContextCallback_object + 85
	54  ???                                 0x02ca17dc 0x0 + 46798812
	55  ???                                 0x02ca1574 0x0 + 46798196
	56  ???                                 0x02ca0cff 0x0 + 46796031
	57  ???                                 0x02c9e5b8 0x0 + 46785976
	58  ???                                 0x007c3564 0x0 + 8140132
	59  ???                                 0x007c1064 0x0 + 8130660
	60  ???                                 0x007c1152 0x0 + 8130898
	61  mono                                0x000c3e4b mono_jit_runtime_invoke + 843
	62  mono                                0x0029e31a do_runtime_invoke + 74
	63  mono                                0x002a1a32 do_exec_main_checked + 146
	64  mono                                0x002a06b5 mono_runtime_run_main_checked + 69
	65  mono                                0x0012ef11 mono_jit_exec + 321
	66  mono                                0x0013218b mono_main + 11627
	67  mono                                0x000b2a57 main + 455
	68  libdyld.dylib                       0xa152f395 start + 1
```

### Logs
https://jenkins.mono-project.com/job/test-mono-pull-request-i386-osx/9088/consoleFull#-2001780300c04ca618-a109-447b-99e0-481aaedae630

https://jenkins.mono-project.com/job/test-mono-pull-request-i386-osx/9104/parsed_console/log.html

**Comment by akoeplinger** [comment]
It's not really crashing in GetHostByName_internal here, it ran into the 5 minutes timeout. I think we only saw this on the on-premises (Boston) bots so it could be a DNS issue there.

**Comment by akoeplinger** [comment]
I assume this is/was fixed by https://github.com/mono/mono/pull/8173

**Comment by akoeplinger** [comment]
I believe we've not seen this particular issue anymore so I'm assuming it was indeed fixed by #8173
